ui/pages/destinations/add.js

```diff
@@ -66,17 +69,24 @@ export default function DestinationsAdd() {
     setSubmitted(true)
   }
 
-  const server = window.location.host
-  let command = `helm install infra-connector infrahq/infra \\
-    --set connector.config.accessKey=${accessKey} \\
-    --set connector.config.server=${server} \\
-    --set connector.config.name=${name}`
+  const server = baseDomain ? `api.${baseDomain}` : window.location.host
+  const values = {
+    connector: {
+      config: {
+        accessKey: accessKey,
+        server: server,
+        name: name,
+      },
+    },
+  }
 
   if (window.location.protocol !== 'https:') {
-    command += ` \\
-    --set connector.config.skipTLSVerify=true`
+    values.connector.config.skipTLSVerify = true
   }
 
+  const valuesYaml = yaml.dump(values)
+  const command = `cat <<EOF | helm upgrade --install infra-connector infrahq/infra -f-\n${valuesYaml}EOF`
```

Comment:
Can we just do this all with flags? The `cat` piping to get multi-line input looks confusing imo

Reply:
We could. It would look like the previous command snippet. IMO it's actually harder to read and understand. Flags simulate a hierarchy which can easily be overlooked. This makes the relationship between the values clear while also making it easier to edit or append.

Additionally, the settings doesn't translate directly to a values file so if the user wants to adds the outputs to IAC, they'll need to transcribe it into YAML. This way, we can add an option to download the file instead of using a heredoc and the `helm` command will remain the same.